src/test/custom_nodes/node_choose_maximum.cpp

```diff
@@ -0,0 +1,163 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include "../../custom_node_interface.h"
+
+static const size_t dummyInputSize = 10;
+
+extern "C" {
+enum METHOD {
+    MAXIMUM_MINIMUM,
+    MAXIMUM_AVERAGE,
+    MAXIMUM_MAXIMUM,
+    METHOD_COUNT
+};
+
+static const std::string INPUT_TENSOR_NAME = "input_tensors";
+
+int execute(const struct CustomNodeTensor* inputs, int inputsLength, struct CustomNodeTensor** outputs, int* outputsLength, const struct CustomNodeParam* params, int paramsLength) {
+    // choose selection criteria
+    uint16_t selection_method = METHOD_COUNT;
+    if (paramsLength != 1) {
+        std::cout << "Wrong number of parameters - expected 1" << std::endl;
+    }
+    for (int i = 0; i < paramsLength; i++) {
```

Comment:
Since we check if paramsLength is equal to 1 in line 41 we do not need to iterate over parameters. You can trust OVMS there is one parameter and access it directly without for loop.

Reply:
All you need to do is to ensure that the only one param passed is the one with correct name - `selection_criterium`.

Reply:
Selection criterium -> selection criteria

Reply:
> All you need to do is to ensure that the only one param passed is the one with correct name - selection_criterium.
That is done in L45